README.md

```diff
@@ -84,31 +86,63 @@ If a single key with in a task needs an update, then all the keys in this task w
   tasks:
     - name: set example values
       elektra:
-        # Despite the name, this does not have to be a real mountpoint in the KDB.
-        # If no plugin is specified, the value is only used as the base key for the keys object.
-        mountpoint: user:/tests/ansible
-        # The keys object contains the data that should be present below the mountpoint.
+        mount:
+          - mountpoint: system:/hosts
+            file: /tmp/hosts
+            includeRecommendedPlugins: true
+            plugins:
+              - hosts:
+          - mountpoint: user:/tests/ansible
+            file: ansible.toml
+            plugins:
+              - toml:
         keys:
-          fruit:
-            cherry:
-              value: cola
-            apple:
-              value: pie
-            berries:
-                raspberry:
-                  value: pi
-                blueberry:
-                  value: muffin
-          vegtables:
-            tomato:
-              value: ketchup
-            potato:
-              value: fries
-
+          - system:
+              hosts:
+                ipv4:
+                  libelektra.org: 1.2.3.4
+          - user:/tests/ansible:
+              fruit:
+                cherry: cola
+                apple: pie
+                berries:
+                    raspberry: pi
+                    blueberry: muffin
+              vegtables:
+                tomato: ketchup
+                potato: fries
 ```
 
-You can also specify metadata (if the underlying storage format supports it).
-In most cases, you should use a specification for metadata instead.
+## Metakeys, Arrays and Subkeys
+
+For every key you can specify metadata, array values and subkeys.
+To do this, you can specify a list with one or more of the following entries directly into the key:
+- `meta`: meta keys of the keys.
+- `array`: Use a TOML array to specify an Elektra array.
+           If you want to set values and metadata on the array index directly, you can use the special name `'#'`.
+           See the `pets` and `friends` keys in the example below.
+- `value`: the value of the key. 
+           Same as if you specify `cherry: cola` in the example above.
+- `keys`: if this is a parent key of child keys, you can continue specifiyng the child keys below this.
+          Obviously this is only needed if you use `value`, `meta` or `array`.
```

Comment:
What do you mean?
like 

```yaml
elektra:
  keys:
    - keys: ....
```
?

This would generate an error, as "keys:" is not a valid namespace.